modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+        }
+    }
+
+    /**
+     * Ensures that {@link Statement#executeBatch()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testBatch() throws Exception {
+        try (Statement stmt = conn.createStatement()) {
+            checkUpdate(5, 10, (off, cnt) -> {
+                for (int i = off; i < off + cnt; i++) {
+                    stmt.addBatch(String.format(SQL_INSERT, i, "name-" + i));
+                }
+
+                stmt.executeBatch();
+            });
+        }
+    }
+
+    /**
+     * Ensures that {@link PreparedStatement#executeBatch()} supports explicit transaction.
+     *
+     * <p>It is expected that in non auto-commit mode, data updates can be rolled back and
+     * are invisible outside of an active transaction until they are committed.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testBatchPrepared() throws Exception {
+        try (PreparedStatement pstmt = conn.prepareStatement(SQL_INSERT_PREPARED)) {
+            checkUpdate(5, 10, (off, cnt) -> {
+                for (int i = off; i < off + cnt; i++) {
+                    pstmt.setInt(1, i);
+                    pstmt.setString(2, "name-" + i);
+
+                    pstmt.addBatch();
+                }
+
+                pstmt.executeBatch();
+            });
+        }
+    }
+
+    /**
+     * Ensures that data updates made in an explicit transaction can be rolled back and are
+     * not visible outside of an active transaction until they are committed.
+     *
+     * @param ops Number of operations.
+     * @param rowsPerOp Number of rows per operation.
+     * @param batchOp Update operation.
+     * @throws SQLException If failed.
+     */
+    private void checkUpdate(int ops, int rowsPerOp, BiConsumerX<Integer, Integer> batchOp) throws SQLException {
+        conn.setAutoCommit(false);
+
+        // Check rollback.
+        checkTxResult(ops, rowsPerOp, batchOp, false);
+
+        // Check commit.
+        checkTxResult(ops, rowsPerOp, batchOp, true);
+    }
+
+    /**
+     * Performs update operations and checks the result of committing or rolling back a transaction.
+     *
+     * @param ops Number of operations.
+     * @param rowsPerOp Number of rows per operation.
+     * @param batchOp Update operation.
+     * @param commit {@code True} to check transaction commit, {@code false} to check rollback.
+     * @throws SQLException If failed.
+     */
+    private void checkTxResult(int ops, int rowsPerOp, BiConsumerX<Integer, Integer> batchOp, boolean commit) throws SQLException {
+        // Creating another connection to ensure that the changes are not visible outside of the transaction.
+        try (Connection conn0 = DriverManager.getConnection(URL)) {
```

Comment:
It took some time to realise that conn0 is not the connection used to make an update... Let's rework this method to make it obvious

Reply:
Thanks, reworked a bit.